client.rb

```diff
@@ -27,16 +27,35 @@ def initialize
     @option = gets.chomp
   end
 
+  def validate_input(message, validation = ->(input) do input != '' end)
+    loop do
+      print message
+      input = gets.chomp
+      puts 'Invalid Input' unless validation.call(input)
+      return input if validation.call(input)
+    end
+  end
+
   def book_info
     # this function is not implemented yet
     # you can ask the user to input data and then return it as on object
     # so the App class will use to add the data
   end
 
   def album_info
-    # this function is not implemented yet
-    # you can ask the user to input data and then return it as on object
-    # so the App class will use to add the data
+    # get the publish date
+    date_regex = %r{^\d{4}/\d{2}/\d{2}$}
+    message = 'Publish date (yyyy/mm/dd): '
+    publish_date = validate_input(message, ->(input) do input.match(date_regex) end)
+
+    # on spotify?
+    message = 'Is on Spotify: '
```

Comment:
Kindly enhance user experience and ensure clear communication, you may want to consider providing the user with specific options, such as prompting them to enter either "yes" or "no", or providing additional alternative choices.